### ISSUE TITLE
MultiMatchQueryBuilderTests#testToQuery fails reproducibly on 6.x

### ISSUE THREAD

**Issue by jpountz** [elastic/elasticsearch#31093]
This seems to be due to the fact that the `_all` field type doesn't support phrase queries.

./gradlew :server:test -Dtests.seed=69AB33AD38246231 -Dtests.class=org.elasticsearch.index.query.MultiMatchQueryBuilderTests -Dtests.method="testToQuery" -Dtests.security.manager=true -Dtests.locale=shi -Dtests.timezone=Africa/Dar_es_Salaam

```
Suite: org.elasticsearch.index.query.MultiMatchQueryBuilderTests
  1> [2018-06-05T10:48:26,627][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testUnknownObjectException]: before test
  1> [2018-06-05T10:48:26,628][INFO ][o.e.p.PluginsService     ] [class org.elasticsearch.test.AbstractQueryTestCase] no modules loaded
  1> [2018-06-05T10:48:26,628][INFO ][o.e.p.PluginsService     ] [class org.elasticsearch.test.AbstractQueryTestCase] no plugins loaded
  1> [2018-06-05T10:48:26,632][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testUnknownObjectException]: after test
  1> [2018-06-05T10:48:26,633][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testFuzzinessOnNonStringField]: before test
  1> [2018-06-05T10:48:26,635][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testFuzzinessOnNonStringField]: after test
  1> [2018-06-05T10:48:26,636][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testEqualsAndHashcode]: before test
  1> [2018-06-05T10:48:26,638][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testEqualsAndHashcode]: after test
  1> [2018-06-05T10:48:26,639][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testToFuzzyQuery]: before test
  1> [2018-06-05T10:48:26,639][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testToFuzzyQuery]: after test
  1> [2018-06-05T10:48:26,640][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testToQuery]: before test
  1> [2018-06-05T10:48:26,649][INFO ][o.e.i.q.MultiMatchQueryBuilderTests] [testToQuery]: after test
  2> REPRODUCE WITH: ./gradlew :server:test -Dtests.seed=69AB33AD38246231 -Dtests.class=org.elasticsearch.index.query.MultiMatchQueryBuilderTests -Dtests.method="testToQuery" -Dtests.security.manager=true -Dtests.locale=shi -Dtests.timezone=Africa/Dar_es_Salaam
ERROR   0.02s J2 | MultiMatchQueryBuilderTests.testToQuery <<< FAILURES!
   > Throwable #1: java.lang.IllegalArgumentException: Can only use phrase queries on text fields - not on [_all] which is of type [_all]
   >    at __randomizedtesting.SeedInfo.seed([69AB33AD38246231:9E50319349A7A7DB]:0)
   >    at org.elasticsearch.index.mapper.MappedFieldType.phraseQuery(MappedFieldType.java:388)
   >    at org.elasticsearch.index.search.MatchQuery$MatchQueryBuilder.analyzePhrase(MatchQuery.java:364)
   >    at org.apache.lucene.util.QueryBuilder.createFieldQuery(QueryBuilder.java:330)
   >    at org.elasticsearch.index.search.MatchQuery$MatchQueryBuilder.createFieldQuery(MatchQuery.java:409)
   >    at org.elasticsearch.index.search.MatchQuery$MatchQueryBuilder.createPhrasePrefixQuery(MatchQuery.java:418)
   >    at org.elasticsearch.index.search.MatchQuery.parse(MatchQuery.java:296)
   >    at org.elasticsearch.index.search.MultiMatchQuery.parseAndApply(MultiMatchQuery.java:60)
   >    at org.elasticsearch.index.search.MultiMatchQuery.parse(MultiMatchQuery.java:73)
   >    at org.elasticsearch.index.query.MultiMatchQueryBuilder.doToQuery(MultiMatchQueryBuilder.java:825)
   >    at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:98)
   >    at org.elasticsearch.test.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:422)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:564)
   >    at java.base/java.lang.Thread.run(Thread.java:844)
```

**Comment by jpountz** [comment]
I just muted this suite.

**Comment by jimczi** [comment]
Fixed by #31072
